File: src/main/java/com/aladdinworksfivefiftyfive/service/StaticTransferSwitchService.java
package com.aladdinworksfivefiftyfive.service;

import java.util.List;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.domain.Specification;
import org.springframework.http.ResponseEntity;

import com.aladdinworksfivefiftyfive.domain.StaticTransferSwitch;
import com.aladdinworksfivefiftyfive.dto.StaticTransferSwitchDTO;
import com.aladdinworksfivefiftyfive.dto.StaticTransferSwitchSearchDTO;
import com.aladdinworksfivefiftyfive.dto.StaticTransferSwitchPageDTO;
import com.aladdinworksfivefiftyfive.dto.StaticTransferSwitchConvertCriteriaDTO;
import com.aladdinworksfivefiftyfive.service.GenericService;
import com.aladdinworksfivefiftyfive.dto.common.RequestDTO;
import com.aladdinworksfivefiftyfive.dto.common.ResultDTO;
import java.util.List;
import java.util.Optional;





public interface StaticTransferSwitchService extends GenericService<StaticTransferSwitch, Integer> {

	List<StaticTransferSwitch> findAll();

	ResultDTO addStaticTransferSwitch(StaticTransferSwitchDTO staticTransferSwitchDTO, RequestDTO requestDTO);

	ResultDTO updateStaticTransferSwitch(StaticTransferSwitchDTO staticTransferSwitchDTO, RequestDTO requestDTO);

    Page<StaticTransferSwitch> getAllStaticTransferSwitchs(Pageable pageable);

    Page<StaticTransferSwitch> getAllStaticTransferSwitchs(Specification<StaticTransferSwitch> spec, Pageable pageable);

	ResponseEntity<StaticTransferSwitchPageDTO> getStaticTransferSwitchs(StaticTransferSwitchSearchDTO staticTransferSwitchSearchDTO);
	
	List<StaticTransferSwitchDTO> convertStaticTransferSwitchsToStaticTransferSwitchDTOs(List<StaticTransferSwitch> staticTransferSwitchs, StaticTransferSwitchConvertCriteriaDTO convertCriteria);

	StaticTransferSwitchDTO getStaticTransferSwitchDTOById(Integer staticTransferSwitchId);







}

File: aladdinworksfivefiftyfive_ui/src/services/StaticTransferSwitchService.js
import http from "../http-common"; 

class StaticTransferSwitchService {
  getAllStaticTransferSwitchs(searchDTO) {
    console.log(searchDTO)
    return this.getRequest(`/staticTransferSwitch/staticTransferSwitchs`, searchDTO);
  }

  get(staticTransferSwitchId) {
    return this.getRequest(`/staticTransferSwitch/${staticTransferSwitchId}`, null);
  }

  findByField(matchData) {
    return this.getRequest(`/staticTransferSwitch?field=${matchData}`, null);
  }

  addStaticTransferSwitch(data) {
    return http.post("/staticTransferSwitch/addStaticTransferSwitch", data);
  }

  update(data) {
  	return http.post("/staticTransferSwitch/updateStaticTransferSwitch", data);
  }
  
  uploadImage(data,staticTransferSwitchId) {
  	return http.postForm("/staticTransferSwitch/uploadImage/"+staticTransferSwitchId, data);
  }




	postRequest(url, data) {
		return http.post(url, data);
      };

	getRequest(url, params) {
        return http.get(url, {
        	params: params
        });
    };

}

export default new StaticTransferSwitchService();
